### ISSUE TITLE
Why Docker

### ISSUE THREAD

**Issue by pkill37** [sickcodes/Docker-eyeOS#19]
What is the point of Docker in your solution? Is it just for the convenience of spawning a container for people unfamiliar with QEMU?

You're still running full VM inside. Sharing the kernel resources with a Linux host (where the containers) won't be possible because (a) you're running full QEMU and (b) iOS kernel is not similar to Linux.

Genuinely curious if there are any interesting technical reasons for using Docker. Thanks for your time!

**Comment by newthis** [comment]
Mayb for the convenience of deployment and good portability, you do not need to reinstall the dependencies  every time.

**Comment by ChaoKaiADG** [comment]
Docker can make sure that applications do not pollute the system environment,so docker.

**Comment by pkill37** [comment]
Yeah but this solution is wrapping QEMU over Docker. QEMU alone already solves the aforementioned nice deployment&portability requirements (after all it's a VM which offers even higher isolation).

I'm not sure what the Docker layer is offering given that it doesn't scale very well to run N of the these "containers".

**Comment by Dreaming-Codes** [comment]
> Yeah but this solution is wrapping QEMU over Docker. QEMU alone already solves the aforementioned nice deployment&portability requirements (after all it's a VM which offers even higher isolation).
> 
> I'm not sure what the Docker layer is offering given that it doesn't scale very well to run N of the these "containers".

This project is using a patched version of qemu so docker is used to avoid packaging the modified version of qemu for each distro.
